File: frontend/src/types/scanner.ts
export interface ScannerCandidate {
  symbol: string;
  name: string;
  price: number;
  change_pct: number;
  volume: number;
  trade_value: number;
  vwap: number;
  volume_spike: number;
  spread_bps: number;
  score: number;
  passed: boolean;
  reasons: string[];
}

export interface ScannerResponse {
  universe_size: number;
  candidates: ScannerCandidate[];
}

File: backend/app/schemas/scanner.py
from decimal import Decimal

from pydantic import BaseModel, Field


class ScannerCandidateResponse(BaseModel):
    symbol: str
    name: str = ""
    price: Decimal
    change_pct: Decimal
    volume: int
    trade_value: Decimal
    vwap: Decimal
    volume_spike: Decimal
    spread_bps: Decimal
    score: Decimal
    passed: bool
    reasons: list[str] = Field(default_factory=list)


class ScannerResponse(BaseModel):
    universe_size: int
    candidates: list[ScannerCandidateResponse]
